subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/IdeDependenciesExtractor.java

```diff
@@ -75,7 +75,7 @@
         // can have multiple IDE dependencies with same component identifier (see GRADLE-1622)
         Multimap<ComponentIdentifier, IdeExtendedRepoFileDependency> resolvedDependenciesComponentMap = LinkedHashMultimap.create();
         for (IdeExtendedRepoFileDependency dep : resolvedExternalDependencies(plusConfigurations, minusConfigurations)) {
-            resolvedDependenciesComponentMap.put(toComponentIdentifier(dep.getId()), dep);
+            resolvedDependenciesComponentMap.put(dep.getComponentIdentifier(), dep);
```

Comment:
I guess `toComponentIdentifier()` can be removed.

Reply:
Yeah, that dies in the next PR :)